cvp_mesh_planner/include/cvp_mesh_planner/cvp_mesh_planner.h

```diff
@@ -162,65 +164,66 @@ class CVPMeshPlanner : public mbf_mesh_core::MeshPlanner
   void computeVectorMap();
 
   /**
-   * @brief Dynamic reconfigure callback
+   * @brief gets called on new incoming reconfigure parameters
+   *
+   * @param cfg new configuration
+   * @param level level
+   * @brief gets called whenever the node's parameters change
+   
+   * @param parameters vector of changed parameters.
+   *                   Note that this vector will also contain parameters not related to the cvp mesh planner
    */
-  void reconfigureCallback(cvp_mesh_planner::CVPMeshPlannerConfig& cfg, uint32_t level);
+  rcl_interfaces::msg::SetParametersResult reconfigureCallback(std::vector<rclcpp::Parameter> parameters);
 
 private:
 
   //! shared pointer to the mesh map
-  mesh_map::MeshMap::Ptr mesh_map;
+  mesh_map::MeshMap::Ptr mesh_map_;
 
   //! the user defined plugin name
-  std::string name;
+  std::string name_;
 
-  //! the private node handle with the user defined namespace (name)
-  ros::NodeHandle private_nh;
+  //! pointer to the node in which this plugin is running
+  rclcpp::Node::SharedPtr node_;
 
   //! flag if cancel has been requested
-  std::atomic_bool cancel_planning;
+  std::atomic_bool cancel_planning_;
 
   //! publisher for the backtracked path
-  ros::Publisher path_pub;
-
-  //! whether to publish the vector field or not
-  bool publish_vector_field;
-
-  //! whether to also publish direction vectors at the triangle centers
-  bool publish_face_vectors;
+  rclcpp::Publisher<nav_msgs::msg::Path>::SharedPtr path_pub_;
 
   //! the map coordinate frame / system id
-  std::string map_frame;
-
-  //! an offset that determines how far beyond the goal (robot's position) is propagated.
-  float goal_dist_offset;
-
-  //! shared pointer to dynamic reconfigure server
-  boost::shared_ptr<dynamic_reconfigure::Server<cvp_mesh_planner::CVPMeshPlannerConfig>> reconfigure_server_ptr;
-
-  //! dynamic reconfigure callback function binding
-  dynamic_reconfigure::Server<cvp_mesh_planner::CVPMeshPlannerConfig>::CallbackType config_callback;
-
-  //! indicates if dynamic reconfigure has been called the first time
-  bool first_config;
-
-  //! the current dynamic reconfigure planner configuration
-  CVPMeshPlannerConfig config;
+  std::string map_frame_;
+
+  // handle of callback for changing parameters dynamically
+  rclcpp::node_interfaces::OnSetParametersCallbackHandle::SharedPtr reconfiguration_callback_handle_;
+  struct {
+    //! whether to publish the vector field or not
+    bool publish_vector_field = false;
+    //! whether to also publish direction vectors at the triangle centers
+    bool publish_face_vectors = false;
+    //! an offset that determines how far beyond the goal (robot's position) is propagated.
+    double goal_dist_offset = 0.3;
+    //! Defines the vertex cost limit with which it can be accessed.
+    double cost_limit = 1.0;
+    //! The vector field back tracking step width.
+    double step_width = 0.4;
+  } config_;
```

Comment:
I did this with member variables in the ftc_controller.
But i like this option, to use a struct member variable with all config options.
Maybe we can change it inside the ftc_controller and use this option for all mbf_plugins